calculation/gmhazard_calc/gmhazard_calc/directivity/directivity.py

```diff
@@ -157,40 +159,63 @@ def compute_fault_directivity(
         constants.EventType.from_rake(rake),
     )
 
-    # Creating the array to store all fdi values
-    fdi_array = np.zeros((n_hypo_data.nhypo, len(sites), len(periods)))
-    phired_array = np.zeros((n_hypo_data.nhypo, len(sites), len(periods)))
-    for index, planes in enumerate(hypo_planes):
-        # Gets the plane index of the hypocentre
-        plane_index = plane_ind[index]
+    if n_procs == 1:
+        fdi, phired = [], []
+        for ix, cur_planes in enumerate(hypo_planes):
+            cur_fdi, (cur_phi_red, cur_predictor_functions, cur_other) = _compute_directivity_effect(
+                lon_lat_depth,
+                cur_planes,
+                plane_ind[ix],
+                sites,
+                nominal_strike,
+                nominal_strike2,
+                mag,
+                rake,
+                periods,
+            )
 
-        fdi, (phi_red, predictor_functions, other) = _compute_directivity_effect(
-            lon_lat_depth,
-            planes,
-            plane_index,
-            sites,
-            nominal_strike,
-            nominal_strike2,
-            mag,
-            rake,
-            periods,
-        )
+            fdi.append(cur_fdi)
+            phired.append(cur_phi_red)
```

Comment:
Previous comment got removed by refresh haha
Is the reason for lists instead of numpy indexing to improve performace as RAM isnt an issue anymore?